modules/fb_instant_articles_display/src/DrupalInstantArticleDisplay.php

```diff
@@ -550,4 +553,71 @@ private function fieldFormatSocialElement($items, $body) {
       $body->addChild($social);
     }
   }
+
+  /**
+   * Add ads if configured in settings
+   */
+  public function addAdsFromSettings() {
+    $ad_type = variable_get('fb_instant_articles_ad_type', AdTypes::FB_INSTANT_ARTICLES_AD_TYPE_NONE);
+    if ($ad_type === AdTypes::FB_INSTANT_ARTICLES_AD_TYPE_NONE) {
+      return;
+    }
+    $dimensions = variable_get('fb_instant_articles_ads_dimension', AdTypes::FB_INSTANT_ARTICLES_AD_TYPE_NONE);
+    $width = 300;
+    $height = 250;
+    switch ($dimensions) {
+      case 0:
+        $width = 300;
+        $height = 250;
```

Comment:
This is kinda odd, and not so future friendly.  What if the options for `$dimensions` followed a convention, like 300x250, which could be broken up to set the `$width` and `$height`.  Easier to define new options, and for a future where maybe we expose an alter hook for the oddball size needs?